src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TimePicker.cs

```diff
@@ -288,10 +289,88 @@ private async Task When_Native_Flyout_Theme(UIKit.UIUserInterfaceStyle expectedS
 			Assert.AreEqual(expectedStyle, nativeTimePicker.OverrideUserInterfaceStyle);
 		}
 #endif
-	}
+#if __IOS__ || __ANDROID__
+		[TestMethod]
+		public async Task When_Time_Uninitialized_Should_Display_Current_Time()
+		{
+			var timePicker = new Microsoft.UI.Xaml.Controls.TimePicker();
+			timePicker.Time = new TimeSpan(-1);
 
-	class MyContext
-	{
-		public object StartTime => null;
+			var expectedCurrentTime = GetCurrentTime();
+
+			TestServices.WindowHelper.WindowContent = timePicker;
+			await TestServices.WindowHelper.WaitForLoaded(timePicker);
+
+			await DateTimePickerHelper.OpenDateTimePicker(timePicker);
+
+			var openFlyouts = FlyoutBase.OpenFlyouts;
+			var associatedFlyout = openFlyouts[0];
+			Assert.IsInstanceOfType(associatedFlyout, typeof(TimePickerFlyout));
+
+#if __ANDROID__
+			var nativeFlyout = (NativeTimePickerFlyout)associatedFlyout;
+
+			var dialogField = typeof(NativeTimePickerFlyout).GetField("_dialog", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
+			var dialog = dialogField?.GetValue(nativeFlyout) as UnoTimePickerDialog;
+			var decorView = dialog.Window?.DecorView;
+
+			var timePickerView = FindTimePicker(decorView);
+
+			var displayedHour = timePickerView.GetHourCompat();
+			var displayedMinute = timePickerView.GetMinuteCompat();
+
+			Assert.AreEqual(expectedCurrentTime.Hours, displayedHour, "Hours should match the current time.");
+			Assert.AreEqual(expectedCurrentTime.Minutes, displayedMinute, "Minutes should match the current time.");
+#elif __IOS__
+		    var nativeFlyout = (NativeTimePickerFlyout)associatedFlyout;
+
+		    var selectorField = typeof(NativeTimePickerFlyout).GetField("_timeSelector", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
+		    var timeSelector = selectorField?.GetValue(nativeFlyout) as TimePickerSelector;
```

Comment:
Let's create a new `internal` method that allows us to retrieve the native dialog/picker and use that instead of reflection